tests/kernel/wave/wave_gemm_test.py

```diff
@@ -87,88 +87,90 @@ def get_test_shapes(test_name: str) -> list[tuple[int]]:
+    b = device_randn(shape[1], shape[2], dtype=datatype)
+    c = device_zeros(shape[0], shape[1], dtype=torch.float32)
+    asm = gemm(a, b, c)
 
-    # With dynamic dimensions, we need to add an assumption on how big
-    # the iterate dimension is to determine whether we can schedule or not.
-    if dynamic_dims:
-        constraints += [tkw.Assumption(K > BLOCK_K * 4)]
+    if dump_generated_mlir:
+        filename = f"wave_gemm_{'x'.join(map(str, shape))}.mlir"
+        with open(filename, "w") as f:
+            f.write(asm)
 
-    # Wave-level micro-kernel.
-    # Since warps are not directly addressable, there is no
-    # explicit notion of a warp id (like a workgroup or thread id).
-    # This kernel uses the input sizes M, N, K throughout, as the tiling
-    # and data movement strategy is determined during the compilation process.
-    # These can be influenced by introducing constraints.
-    @tkw.wave(constraints)
-    def gemm(
-        a: tkl.Memory[M, K, ADDRESS_SPACE, tkl.f16],
-        b: tkl.Memory[N, K, ADDRESS_SPACE, tkl.f16],
-        c: tkl.Memory[M, N, GLOBAL_ADDRESS_SPACE, tkl.f32],
-    ):
-        c_reg = tkl.Register[M, N, tkl.f32](0.0)
+    if run_bench:
+        options.benchmark_results_file = perf_filename_iree
 
-        # This microkernel encodes the fact that if the iterate
-        # dimension were tiled, then we would need to materialize a loop.
-        @tkw.iterate(K, init_args=[c_reg])
-        def repeat(acc: tkl.Register[M, N, tkl.f32]) -> tkl.Register[M, N, tkl.f32]:
-            # a_reg: tkw.Register[M, K, tkl.f16]
-            a_reg = tkw.read(a)
-            # b_reg: tkw.Register[N, K, tkl.f16]
-            b_reg = tkw.read(b)
-            # acc: tkw.Register[M, N, tkl.f32]
-            acc = tkw.mma(a_reg, b_reg, acc)
-            return acc
+    iree_ref = device_zeros(shape[0], shape[1], dtype=torch.float32)
+    generate_iree_ref("mmt", [a, b], [iree_ref])
+    assert_close(c, iree_ref, check_device=False)
 
-        # repeat represents the results of the loop
-        tkw.write(repeat, c)
 
-    hyperparams = {
-        ADDRESS_SPACE: SHARED_ADDRESS_SPACE,
-        BLOCK_M: 64,
-        BLOCK_N: 64,
-        BLOCK_K: 32,
-        M: shape[0],
-        N: shape[1],
-        K: shape[2],
-    }
-    hyperparams.update(get_default_scheduling_params())
+_xfail = lambda *a: pytest.param(*a, marks=pytest.mark.xfail)
 
-    dynamic_symbols = []
-    if dynamic_dims:
-        dynamic_symbols.append(M)
-        dynamic_symbols.append(N)
-        dynamic_symbols.append(K)
-        del hyperparams[M]
-        del hyperparams[N]
-        del hyperparams[K]
+
+@require_e2e
+@require_cdna3_or_4
+@pytest.mark.parametrize("shape", get_test_shapes("test_gemm"))
+@pytest.mark.parametrize(
+    "enable_scheduling",
+    [
+        SchedulingType.NONE,
+        _xfail(SchedulingType.PREFETCH),
+        _xfail(SchedulingType.MODULO),
+        _xfail(SchedulingType.MODULO_MULTI_BUFFERED),
```

Comment:
Are these tests expensive? if yes, thoughts on just adding this as a TODO to fix as opposed to it running here?

Reply:
They should fail (somewhat) quickly, and we will need to support the eventually so let's keep them.